src/main/java/org/opensearch/neuralsearch/search/query/HybridQueryPhaseSearcher.java

```diff
@@ -51,26 +53,28 @@ public boolean searchWith(
         }
     }
 
-    private static boolean hasNestedFieldOrNestedDocs(final Query query, final SearchContext searchContext) {
-        return searchContext.mapperService().hasNested() && new NestedHelper(searchContext.mapperService()).mightMatchNestedDocs(query);
-    }
-
     private static boolean isWrappedHybridQuery(final Query query) {
         return query instanceof BooleanQuery
             && ((BooleanQuery) query).clauses().stream().anyMatch(clauseQuery -> clauseQuery.getQuery() instanceof HybridQuery);
     }
 
     @VisibleForTesting
     protected Query extractHybridQuery(final SearchContext searchContext, final Query query) {
-        if (hasNestedFieldOrNestedDocs(query, searchContext)
+        if ((hasAliasFilter(query, searchContext) || hasNestedFieldOrNestedDocs(query, searchContext))
             && isWrappedHybridQuery(query)
-            && ((BooleanQuery) query).clauses().size() > 0) {
-            // extract hybrid query and replace bool with hybrid query
+            && !((BooleanQuery) query).clauses().isEmpty()) {
             List<BooleanClause> booleanClauses = ((BooleanQuery) query).clauses();
-            if (booleanClauses.isEmpty() || booleanClauses.get(0).getQuery() instanceof HybridQuery == false) {
-                throw new IllegalStateException("cannot process hybrid query due to incorrect structure of top level bool query");
+            if (!(booleanClauses.get(0).getQuery() instanceof HybridQuery)) {
+                throw new IllegalStateException("cannot process hybrid query due to incorrect structure of top level query");
             }
-            return booleanClauses.get(0).getQuery();
+            HybridQuery hybridQuery = (HybridQuery) booleanClauses.stream().findFirst().get().getQuery();
+            List<Query> filterQueries = booleanClauses.stream()
+                .skip(1)
```

Comment:
why this?

Reply:
first clause in this case will be hybrid query, we've checked in this is wrapped hybrid query earlier. the rest of clauses that are of type `filter` we should use as filters and rewrite sub-queries

Reply:
the structure of query is something like this:
```
  bool : {
     must: [
       hybrid_query
     ],
     filter: {
       alias_filter1,
       alias_filter2,
       field_exists (added for nested field)
     }
  }
```
in java that is:
```
  bool : {
     clause1: {
       hybrid_query,
       occur: must
     },
     clause2: {
       alias_filter1,
       occur: filter
     },
     clause3: {
       alias_filter2,
       occur: filter
     },
     clause4: {
       field_exists,
       occur: filter
     }
  }
```

Reply:
can we use this occur to get all the fiters out rather than doing skip(1)

Reply:
it's actually there on the next line `.filter(clause -> BooleanClause.Occur.FILTER == clause.getOccur())`, that skip(1) is like a small optimization that only removes one element that we know isn't a filter for sure. But deleting that `skip` function will not harm, let me create new commit

Reply:
pushed corresponding code change

Reply:
its just good that we don't skip like this.